cpp11/Ice/interceptor/Client.cpp

```diff
@@ -57,93 +57,126 @@ main(int argc, char* argv[])
+
 int
 run(const shared_ptr<Ice::Communicator>& communicator)
 {
+    auto context = communicator->getImplicitContext();
     auto thermostat = Ice::checkedCast<Demo::ThermostatPrx>(communicator->propertyToProxy("Thermostat.Proxy"));
     if(!thermostat)
     {
         cerr << "invalid thermostat proxy" << endl;
         return 1;
     }
 
-    //
-    // Read in values from the command line for the username, password, and
-    // the temperature to try and set the thermostat to.
-    //
-    string username;
-    string password;
-    string temp;
-    float desiredTemp;
-    try
-    {
-        cout << "Username:" << endl;
-        getline(cin, username);
-        cout << "Password:" << endl;
-        getline(cin, password);
-        cout << "Enter desired temperature:" << endl;
-        getline(cin, temp);
-        desiredTemp = stof(temp);
-    }
-    catch(const invalid_argument&)
-    {
-        cerr << "Specified temperature is not a float value." << endl;
-        return 1;
-    }
-
-    //
-    // Tries calling 'setTemp' without authorization and receives an exception.
-    //
-    cout << "Attempting to set temperature without access token..." << endl;
-    try
-    {
-        cout << "Current temperature is " << thermostat->getTemp() << endl;
-        thermostat->setTemp(desiredTemp);
-        cout << "New temperature is " << thermostat->getTemp() << endl;
-    }
-    catch(const Demo::AuthorizationException&)
-    {
-        cout << "Failed to set temperature. Access denied!" << endl;
-    }
-    cout << endl;
-
-    //
-    // Request an access token from the server's authentication object.
-    //
     auto authenticator = Ice::checkedCast<Demo::AuthenticatorPrx>(communicator->propertyToProxy("Authenticator.Proxy"));
     if(!authenticator)
     {
         cerr << "invalid authenticator proxy" << endl;
         return 1;
     }
-    Demo::Token token = authenticator->getToken(username, password);
-    cout << "Successfully retrieved access token: \"" + token.value + "\"" << endl;
 
-    //
-    // Add the access token to the communicator's context, so it will be
-    // sent along with every request made through it.
-    //
-    auto context = communicator->getImplicitContext();
-    context->put("accessToken", token.value);
+    menu();
 
-    //
-    // Tries calling 'setTemp' again, this time with the access token.
-    //
-    cout << "Attempting to set temperature with access token..." << endl;
-    try
-    {
-        cout << "Current temperature is " << thermostat->getTemp() << endl;
-        thermostat->setTemp(desiredTemp);
-        cout << "New temperature is " << thermostat->getTemp() << endl;
-    }
-    catch(const Demo::AuthorizationException&)
+    string line;
+    do
     {
-        //
-        // No longer encountered since the client has authorization to use 'setTemp' now.
-        //
-        cout << "Failed to set temperature. Access denied!" << endl;
+        try
+        {
+            cout << "\n==> ";
```

Comment:
It is better to add `<< flush` after output to`cout` if we are not using `<< endl`, we have previously problems with platforms that do not auto flush the output in non interactive terminals (running the demos in CI from a python script)